File: BackendAPI/Application/DTOs/Output/OrderDetailsOutputDto.cs
using Domain.Enums;

namespace Application.DTOs.Output;

public class OrderDetailsOutputDto
{
    public Guid Id { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public OrderStatus Status { get; set; }
    public DateTimeOffset? ClosedAt { get; set; }
    public required decimal TotalAmount { get; set; }
    public required int TotalItems { get; set; }
    public required List<OrderItemOutputDto> Products { get; set; }
}

File: FrontendWeb/src/declarations/dtos/output/OrderDetailsOutputDto.ts
import { OrderStatus } from '../../enums/OrderStatus';
import { OrderItemOutputDto } from './OrderItemOutputDto';

// OrderDetailsOutputDto.ts
export interface OrderDetailsOutputDto {
	id: string;
	createdAt: string;
	status: OrderStatus;
	closedAt?: string;
	totalAmount: number;
	totalItems: number;
	products: OrderItemOutputDto[];
}
